ark/segmentation/fiber_segmentation.py

```diff
@@ -0,0 +1,298 @@
+        object_properties (Iterable[str]):
+            Properties to compute, any keyword for region props may be used.  Defaults are:
+             - major_axis_length
+             - minor_axis_length
+             - orientation
+             - centroid
+             - label
+             - eccentricity
+             - euler_number
+        save_csv (bool):
+            Whether or not to save csv of fiber objects
+        debug (bool):
+            Save intermediate preprocessing steps
+
+    Returns:
+        pd.DataFrame, Dict[str, np.ndarray]:
+         - Dataframe containing the fiber objects and their properties
+         - Dictionary mapping fov names to their fiber label images
+    """
+    channel_xr = data_xr.loc[:, :, :, fiber_channel]
+    fov_len = channel_xr.shape[1]
+
+    fiber_label_images = {}
+    fiber_object_table = []
+
+    if debug:
+        debug_path = os.path.join(out_dir, '_debug')
+        if not os.path.exists(debug_path):
+            os.makedirs(debug_path)
+
+    for fov in channel_xr.fovs.values:
+        fiber_channel_data = channel_xr.loc[fov, :, :].values.astype('float')
+        blurred = ndi.gaussian_filter(fiber_channel_data, sigma=blur)
+
+        # local contrast enhancement
+        contrast_adjusted = equalize_adapthist(
+            blurred / np.max(blurred),
+            kernel_size=fov_len / contrast_scaling_divisor
+        )
+
+        # meijering filtering
+        ridges = meijering(contrast_adjusted, sigmas=fiber_widths, black_ridges=False)
+
+        # remove image intensity influence for watershed setup
+        distance_transformed = ndi.gaussian_filter(
+            distance_transform_edt(ridges > ridge_cutoff),
+            sigma=1
+        )
+
+        # watershed setup
+        threshed = np.zeros_like(distance_transformed)
+        thresholds = threshold_multiotsu(distance_transformed, classes=3)
+
+        threshed[distance_transformed < thresholds[0]] = 1
+        threshed[distance_transformed > thresholds[1]] = 2
+
+        elevation_map = sobel(
+            ndi.gaussian_filter(distance_transformed, sigma=sobel_blur)
+        )
+
+        segmentation = watershed(elevation_map, threshed) - 1
+
+        labeled, _ = ndi.label(segmentation)
+
+        labeled_filtered = remove_small_objects(labeled, min_size=min_fiber_size) * segmentation
+
+        if debug:
+            imsave(os.path.join(debug_path, f'{fov}_thresholded.tiff'), threshed,
+                   check_contrast=False)
+            imsave(os.path.join(debug_path, f'{fov}_ridges_thresholded.tiff'),
+                   distance_transformed, check_contrast=False)
+            imsave(os.path.join(debug_path, f'{fov}_meijering_filter.tiff'), ridges,
+                   check_contrast=False)
+            imsave(os.path.join(debug_path, f'{fov}_contrast_adjusted.tiff'), contrast_adjusted,
+                   check_contrast=False)
+
+        imsave(os.path.join(out_dir, f'{fov}_fiber_labels.tiff'), labeled_filtered,
```

Comment:
Not sure if this is supposed to be indented or not. If not, why do we also return the dictionary?

Reply:
Nope, the fiber label images should always be outputted. 
Per Adam's notebook documentation: "The function saves the fiber object table as a csv, and the labeled images as tiff files. The function also returns these as variables, for continued processing/exploration by the user."

Not sure what exactly the plan was, but it you think that's note something users will need I can remove the return.